lib/Monad_WP/OptionMonad.thy

```diff
@@ -48,14 +68,115 @@ lemma opt_map_upd_Some:
   "f(x \<mapsto> v) |> g = (f |> g)(x := g v)"
   by (auto simp: opt_map_def)
 
-lemmas opt_map_upd[simp] = opt_map_upd_None opt_map_upd_Some
+lemma opt_map_Some_upd_Some:
+  "f(x \<mapsto> v) ||> g = (f ||> g)(x \<mapsto> g v)"
+  by (simp add: opt_map_upd_Some)
+
+lemmas opt_map_upd[simp]
+  = opt_map_upd_None opt_map_upd_Some opt_map_Some_upd_Some
+
+lemma opt_map_upd_triv:
+  "t k = Some x \<Longrightarrow> (t |> f)(k := f x) = t |> f"
```

Comment:
not sure where "t k" terminology came from

Reply:
probably from the application area. `k` makes sense as a generic key, `t` I'm not sure about. `g` would be slightly better, but if we do that we should do it in both branches.